addons/im_livechat/models/res_partner.py

```diff
@@ -9,6 +9,32 @@ class Partners(models.Model):
 
     user_livechat_username = fields.Char(compute='_compute_user_livechat_username')
 
+    @api.model
+    def search_for_channel_invite(self, search_term, channel_id=None, limit=30):
+        result = super().search_for_channel_invite(search_term, channel_id, limit)
+        channel = self.env['discuss.channel'].browse(channel_id)
+        partners = self.browse([partner["id"] for partner in result['partners']])
+        if channel.channel_type != 'livechat' or not partners:
+            return result
```

Comment:
When channel is not provided, won't this return not trigger?

Reply:
I think it would work: 

- `channel.channel_type` => False on an empty recordset.
- `False != 'livechat` => the original result is returned

Am i missing something?